test/app-luatest/gh_7000_compat_module_test.lua

```diff
@@ -0,0 +1,262 @@
+    for _, policy in pairs(policies) do
+        arg[policy.name] = 'old'
+    end
+    compat(arg)
+    for _, policy in pairs(policies) do
+        t.assert_equals(compat[policy.name].current, OLD)
+    end
+    test_options_calls()
+end
+
+g.test_serialize = function()
+    reset(compat)
+    local serialized = getmetatable(compat).__serialize(compat)
+
+    local prev_pos = 0
+    local combined = {}
+    for pos, option in pairs(serialized) do
+        t.assert_equals(pos, prev_pos + 1)
+        prev_pos = pos
+
+        local name, value = pairs(option)(option)
+        combined[name] = value
+    end
+
+    local res = { }
+    for _, policy in pairs(policies) do
+        res[policy.name] = ('default (%s)'):format(policy.default)
+    end
+    t.assert_items_include(combined, res)
+end
+
+g.test_dump = function()
+    reset(compat)
+    assert(compat.option_1.current == DEFAULT)
+    assert(loadstring(compat.dump()))()
+    t.assert_equals(compat.option_1.current, DEFAULT)
+    assert(loadstring(compat.dump('new')))()
+    t.assert_equals(compat.option_1.current, NEW)
+    assert(loadstring(compat.dump('old')))()
+    t.assert_equals(compat.option_1.current, OLD)
+    assert(loadstring(compat.dump('default')))()
+    t.assert_equals(compat.option_1.current, DEFAULT)
+    assert(loadstring(compat.dump('current')))()
+    t.assert_equals(compat.option_1.current, NEW)
+    reset(compat)
+end
+
+g.test_help = function()
+    t.assert(compat.help())
+end
+
+g.test_hot_reload = function()
+    local hot_reload_policy_1 = {
+        name = 'hot_reload_option',
+        default = 'new',
+        brief = 'hot_reload_option_1',
+        obsolete = nil,
+        action = function()
+            hot_reload_option_called = true
+        end,
+        run_action_now = true
+    }
+
+    compat.add_option(hot_reload_policy_1)
+    compat.hot_reload_option = 'new'
+    t.assert_equals(compat.hot_reload_option.current, NEW)
+    t.assert(hot_reload_option_called)
+    hot_reload_option_called = false
+
+    local hot_reload_policy_2 = {
+        name = 'hot_reload_option',
+        default = 'old',
+        brief = 'hot_reload_option_2',
+        obsolete = nil,
+        action = function()
+            hot_reload_option_called = true
+        end
+        -- run_action_now == false by default.
+    }
+
+    compat.add_option(hot_reload_policy_2)
+    t.assert_equals(compat.hot_reload_option.current, NEW)
+    t.assert_equals(compat.hot_reload_option.brief, 'hot_reload_option_2')
+    t.assert_not(hot_reload_option_called)
+end
+
+-- Autocomplete tests.
+
+local console = require('console')
+local server  = require('test.luatest_helpers.server')
+
+local function tabcomplete(s)
+    return console.completion_handler(s, 0, #s)
+end
+
+g.test_autocomplete = function()
+    g.server = server:new({alias = 'master'})
+    g.server:start()
+    g.server:exec(function()
+        box.schema.create_space('space1'):create_index('primary')
+    end)
```

Comment:
Is it not used anywhere? Or I miss something?

Reply:
You are right, left from the old times. Fixed.

Reply:
You still require the `server` helper. I suggest to check the test using luatest, it shows such dead variables.